mp4parse/src/lib.rs

```diff
@@ -4058,13 +4070,31 @@ fn read_meta<T: Read>(src: &mut BMFFBox<T>) -> Result<MetadataBox> {
     while let Some(mut b) = iter.next_box()? {
         match b.head.name {
             BoxType::MetadataItemListEntry => read_ilst(&mut b, &mut meta)?,
+            BoxType::MetadataXMLBox => read_xml_(&mut b, &mut meta)?,
+            BoxType::MetadataBXMLBox => read_bxml(&mut b, &mut meta)?,
             _ => skip_box_content(&mut b)?,
         };
         check_parser_state!(b.content);
     }
     Ok(meta)
 }
 
+/// Parse a XML box inside a meta box
+fn read_xml_<T: Read>(src: &mut BMFFBox<T>, meta: &mut MetadataBox) -> Result<()> {
+    let (_, _) = read_fullbox_extra(src)?;
+    let size = src.content.limit();
+    meta.xml = Some(read_buf(&mut src.content, size)?);
```

Comment:
This can be done a bit more concisely using the convenience function in `fallible_collections` (and assuming the structure change I suggested above):
```suggestion
    meta.xml = Some(XmlBox::StringXmlBox(src.read_into_try_vec()?));
```

Similarly for `read_bxml`.